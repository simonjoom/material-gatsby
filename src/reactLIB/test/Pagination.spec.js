import React from 'react';
import { shallow, mount } from 'enzyme';
import Pagination from '../src/Pagination';

describe('<Pagination />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Pagination items={10} activePage={2} maxButtons={6} />);
  });

  test('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('has items', () => {
    expect(wrapper.children()).toHaveLength(8);
  });

  test('accepts activePage prop', () => {
    expect(wrapper.childAt(2).props().active).toBeTruthy();
  });

  test('accepts active prop', () => {
    wrapper.setState({ activePage: 4 });
    expect(wrapper).toMatchSnapshot();
  });

  test("doesn't update state with wrong activePage prop", () => {
    wrapper = shallow(<Pagination items={1} activePage={2} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('updates state with passed activePage prop', () => {
    const newActive = 5;
    wrapper = mount(<Pagination items={10} activePage={2} />);
    wrapper.setProps({ activePage: newActive });

    expect(wrapper.state().activePage).toEqual(newActive);
  });
});
