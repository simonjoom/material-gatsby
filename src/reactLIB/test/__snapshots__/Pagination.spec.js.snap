// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> accepts active prop 1`] = `
<ul
  className="pagination"
>
  <PaginationButton
    disabled={false}
    key="pagination-0"
    onSelect={[Function]}
  >
    <Icon>
      chevron_left
    </Icon>
  </PaginationButton>
  <PaginationButton
    active={false}
    key="pagination-1"
    onSelect={[Function]}
  >
    1
  </PaginationButton>
  <PaginationButton
    active={false}
    key="pagination-2"
    onSelect={[Function]}
  >
    2
  </PaginationButton>
  <PaginationButton
    active={false}
    key="pagination-3"
    onSelect={[Function]}
  >
    3
  </PaginationButton>
  <PaginationButton
    active={true}
    key="pagination-4"
    onSelect={[Function]}
  >
    4
  </PaginationButton>
  <PaginationButton
    active={false}
    key="pagination-5"
    onSelect={[Function]}
  >
    5
  </PaginationButton>
  <PaginationButton
    active={false}
    key="pagination-6"
    onSelect={[Function]}
  >
    6
  </PaginationButton>
  <PaginationButton
    disabled={false}
    key="pagination-11"
    onSelect={[Function]}
  >
    <Icon>
      chevron_right
    </Icon>
  </PaginationButton>
</ul>
`;

exports[`<Pagination /> doesn't update state with wrong activePage prop 1`] = `
<ul
  className="pagination"
>
  <PaginationButton
    disabled={true}
    key="pagination-0"
    onSelect={[Function]}
  >
    <Icon>
      chevron_left
    </Icon>
  </PaginationButton>
  <PaginationButton
    active={true}
    key="pagination-1"
    onSelect={[Function]}
  >
    1
  </PaginationButton>
  <PaginationButton
    disabled={true}
    key="pagination-2"
    onSelect={[Function]}
  >
    <Icon>
      chevron_right
    </Icon>
  </PaginationButton>
</ul>
`;

exports[`<Pagination /> renders 1`] = `
<ul
  className="pagination"
>
  <PaginationButton
    disabled={false}
    key="pagination-0"
    onSelect={[Function]}
  >
    <Icon>
      chevron_left
    </Icon>
  </PaginationButton>
  <PaginationButton
    active={false}
    key="pagination-1"
    onSelect={[Function]}
  >
    1
  </PaginationButton>
  <PaginationButton
    active={true}
    key="pagination-2"
    onSelect={[Function]}
  >
    2
  </PaginationButton>
  <PaginationButton
    active={false}
    key="pagination-3"
    onSelect={[Function]}
  >
    3
  </PaginationButton>
  <PaginationButton
    active={false}
    key="pagination-4"
    onSelect={[Function]}
  >
    4
  </PaginationButton>
  <PaginationButton
    active={false}
    key="pagination-5"
    onSelect={[Function]}
  >
    5
  </PaginationButton>
  <PaginationButton
    active={false}
    key="pagination-6"
    onSelect={[Function]}
  >
    6
  </PaginationButton>
  <PaginationButton
    disabled={false}
    key="pagination-11"
    onSelect={[Function]}
  >
    <Icon>
      chevron_right
    </Icon>
  </PaginationButton>
</ul>
`;
